### PR TITLE
Fix env var commands

### PR DESCRIPTION
**Pull request by cianclarke**
Old env var commands no longer function with our updates to multiple environments. Updates to use correct APIs.

### CODE REVIEW

File: lib/cmd/common/initlocal.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ var fs = require('fs');
 var url = require('url');
 var exec = require('child_process').exec;
 var cheerio = require('cheerio');
-var env = require('./env.js');
+var _ = require('underscore');
```
Comment (cianclarke) [inline lib/cmd/common/initlocal.js]: I'd imagine nobody is using `initlocal` any more, but just incase, here's some backwards compatibility. Fixing the references was easier than ripping the command...

File: lib/cmd/fh3/app/envvars/list.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    0 : 'app',
+    1 : 'env'
+  },
+  'describe' : {
+    'app' : 'Unique 24 character GUID of the app you want to retrieve env vars from',
+    'env' : 'Environment to look up the env vars for',
+    'deployed' : 'Only list user created environment variables? These have typically already been deployed to an environment'
+  },
+  'url' : function(argv){
+    if (argv.deployed){
```
Comment (maleck13) [inline lib/cmd/fh3/app/envvars/list.js]: Is this right? so if we want the deployed env vars we call millicore but if we want them all we call supercore. Seems unfortunate if this is the case

File: lib/cmd/fh3/app/envvars/create.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    'app': 'a',
+    'env': 'e'
+  },
+  'describe': {
+    'app' : 'Unique 24 character GUID of the app you want to retrieve env vars from',
+    'env' : 'Environment to look up the env vars for',
+    'name': 'The name, or key of your environment variable - this will be accessible by process.env.$NAME',
+    'value': 'The value assigned to this env var'
+  },
+  'url' : function(argv){
```
Comment (maleck13) [inline lib/cmd/fh3/app/envvars/create.js]: What is the value being posted here, it is not clear to me. Is it the contents of the  cli flags?

File: lib/cmd/fh3/app/envvars/unset.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  },
+  'url' : function(argv){  
+    return "/box/api/apps/" + fhc.appId(argv.app) + "/env/" + argv.env + "/envvars/unset";  
+  },
+  preCmd : function(params, cb){
+    var payload = [params.id];
+    payload.id = params.id;
+    payload.app = params.app;
+    payload.env = params.env;
+    // payload is an array of IDs to unset..
```
Comment (maleck13) [inline lib/cmd/fh3/app/envvars/unset.js]: so payload is assigned as an array, why do we assign values to the array object like this? Should it not be

```
payload[params.id] = payload.id 

```

File: lib/cmd/fh3/app/envvars/list.js (hunk 2 above)
Comment (cianclarke) [inline lib/cmd/fh3/app/envvars/list.js]: Yep - "deployed" is the env vars that have been set in the studio (via millicore). The mBaaS endpoint shows every environment variable, including the FH_ ones. 
Env var endpoints are a bit of a disaster, full of bits of legacy shite that is in theory longer used  :-(

File: lib/cmd/fh3/app/envvars/create.js (hunk 3 above)
Comment (cianclarke) [inline lib/cmd/fh3/app/envvars/create.js]: Yep - default behavior of commands is to send a JSON-encoded representation of the params being passed in, with matching keys as per "describe"

File: lib/cmd/fh3/app/envvars/unset.js (hunk 4 above)
Comment (cianclarke) [inline lib/cmd/fh3/app/envvars/unset.js]: Really just a workaround way of passing the App ID, env name and env var ID to the 'url' function on 22, but not have them sent in the outbound payload.
When this "arrayish" thing is JSON.stringified, only the array[] part gets POSTed

**Comment by wtrocki** [comment]
@cianclarke are you still interested in getting this into fh-fhc?

**Comment by david-martin** [comment]
@cianclarke Unfortunate that this wasn't merged once approved.
I'm going to close it for now as its out of date and need rebasing.
If you want to get it up to date and reopen, I'd be more than happy to help get it merged.
RE: initlocal, i'd say ignore that completely. `fhc local` has not been the recommended way of doing local dev in a long time, with `grunt serve` or `npm run dev` being a more common thing these days

**Pull request closed without merging**